backend/modules/parsers/unstructured_io.py

```diff
@@ -0,0 +1,92 @@
+import requests
+from langchain.docstore.document import Document
+from requests.adapters import HTTPAdapter, Retry
+
+from backend.logger import logger
+from backend.modules.parsers.parser import BaseParser
+from backend.settings import settings
+
+
+class UnstructuredIoParser(BaseParser):
+    """
+    UnstructuredIoParser is a parser class for extracting text from unstructured input.
+    """
+
+    supported_file_extensions = [
+        ".txt",
+        ".eml",
+        ".msg",
+        ".xml",
+        ".html",
+        ".md",
+        ".rst",
+        ".json",
+        ".rtf",
+        ".jpeg",
+        ".png",
+        ".doc",
+        ".docx",
+        ".ppt",
+        ".pptx",
+        ".pdf",
+        ".odt",
+        ".epub",
+        ".csv",
+        ".tsv",
+        ".xlsx",
+    ]
+
+    def __init__(self, max_chunk_size: int = 2000, *args, **kwargs):
+        """
+        Initializes the UnstructuredIoParser object.
+        """
+        self.max_chunk_size = max_chunk_size
+        self.session = requests.Session()
+        self.retry_strategy = Retry(
+            total=3,
+            backoff_factor=1,
+            status_forcelist=[429, 500, 502, 503, 504],
+            allowed_methods=["POST"],
+        )
+        self.adapter = HTTPAdapter(max_retries=self.retry_strategy)
+        self.session.mount("https://", self.adapter)
+        self.session.mount("http://", self.adapter)
+        self.headers = {
+            "accept": "application/json",
+        }
```

Comment:
All this retrying should be made a common utility across the repo

Reply:
will take this up as a separate PR